File: sdks/php/src/AddOns/Requests/ApplyAddOnDto.php
<?php

namespace NovaBilling\AddOns\Requests;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Core\Json\JsonProperty;

class ApplyAddOnDto extends JsonSerializableType
{
    /**
     * @var string $addOnId Add-on ID
     */
    #[JsonProperty('addOnId')]
    public string $addOnId;

    /**
     * @var string $customerId Customer ID
     */
    #[JsonProperty('customerId')]
    public string $customerId;

    /**
     * @var ?string $subscriptionId Subscription to attach the charge to
     */
    #[JsonProperty('subscriptionId')]
    public ?string $subscriptionId;

    /**
     * @var float $amount Charge amount
     */
    #[JsonProperty('amount')]
    public float $amount;

    /**
     * @var string $currency Currency
     */
    #[JsonProperty('currency')]
    public string $currency;

    /**
     * @param array{
     *   addOnId: string,
     *   customerId: string,
     *   amount: float,
     *   currency: string,
     *   subscriptionId?: ?string,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->addOnId = $values['addOnId'];
        $this->customerId = $values['customerId'];
        $this->subscriptionId = $values['subscriptionId'] ?? null;
        $this->amount = $values['amount'];
        $this->currency = $values['currency'];
    }
}

File: sdks/csharp/src/NovaBilling/AddOns/Requests/ApplyAddOnDto.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record ApplyAddOnDto
{
    /// <summary>
    /// Add-on ID
    /// </summary>
    [JsonPropertyName("addOnId")]
    public required string AddOnId { get; set; }

    /// <summary>
    /// Customer ID
    /// </summary>
    [JsonPropertyName("customerId")]
    public required string CustomerId { get; set; }

    /// <summary>
    /// Subscription to attach the charge to
    /// </summary>
    [JsonPropertyName("subscriptionId")]
    public string? SubscriptionId { get; set; }

    /// <summary>
    /// Charge amount
    /// </summary>
    [JsonPropertyName("amount")]
    public required double Amount { get; set; }

    /// <summary>
    /// Currency
    /// </summary>
    [JsonPropertyName("currency")]
    public required string Currency { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
